File: frontend/src/types/analytics.ts
/**
 * Analytics, auto-sync, and notification type definitions.
 */

export interface MonthlyTurnover {
  month: string;
  issuedCount: number;
  issuedNet: number;
  issuedGross: number;
  receivedCount: number;
  receivedNet: number;
  receivedGross: number;
}

export interface InvoiceTypeSummary {
  invoiceType: string;
  count: number;
  netTotal: number;
  vatTotal: number;
  grossTotal: number;
}

export interface TopCounterparty {
  nip: string;
  name: string;
  invoiceCount: number;
  totalGross: number;
}

export interface AnalyticsSummary {
  invoiceCount: number;
  totalIssued: number;
  totalReceived: number;
  monthlyTurnover: MonthlyTurnover[];
  typeSummary: InvoiceTypeSummary[];
  topIssuedCounterparties: TopCounterparty[];
  topReceivedCounterparties: TopCounterparty[];
}

export interface AutoSyncConfig {
  enabled: boolean;
  intervalMinutes: number;
  syncIssued: boolean;
  syncReceived: boolean;
  lastRunAt?: string;
}

export interface NotificationConfig {
  enabled: boolean;
  webhookUrl?: string;
  webhookEnabled: boolean;
  emailEnabled: boolean;
  emailTo?: string;
  smtpHost?: string;
  smtpPort: number;
  smtpUsername?: string;
  smtpPassword?: string;
  smtpUseTls: boolean;
}

export interface NotificationHistoryEntry {
  id: number;
  notificationType: string;
  triggerEvent: string;
  recipient?: string;
  status: string;
  errorMessage?: string;
  createdAt: string;
  sentAt?: string;
}

File: backend/app/models/analytics.py
"""Analytics and reporting models."""

from typing import List, Optional
from pydantic import BaseModel, Field


class MonthlyTurnover(BaseModel):
    month: str
    issued_count: int = Field(alias="issuedCount")
    issued_net: float = Field(alias="issuedNet")
    issued_gross: float = Field(alias="issuedGross")
    received_count: int = Field(alias="receivedCount")
    received_net: float = Field(alias="receivedNet")
    received_gross: float = Field(alias="receivedGross")

    class Config:
        populate_by_name = True


class InvoiceTypeSummary(BaseModel):
    invoice_type: str = Field(alias="invoiceType")
    count: int
    net_total: float = Field(alias="netTotal")
    vat_total: float = Field(alias="vatTotal")
    gross_total: float = Field(alias="grossTotal")

    class Config:
        populate_by_name = True


class TopCounterparty(BaseModel):
    nip: str
    name: str
    invoice_count: int = Field(alias="invoiceCount")
    total_gross: float = Field(alias="totalGross")

    class Config:
        populate_by_name = True


class AnalyticsSummary(BaseModel):
    invoice_count: int = Field(alias="invoiceCount")
    total_issued: float = Field(alias="totalIssued")
    total_received: float = Field(alias="totalReceived")
    monthly_turnover: List[MonthlyTurnover] = Field(alias="monthlyTurnover")
    type_summary: List[InvoiceTypeSummary] = Field(alias="typeSummary")
    top_issued_counterparties: List[TopCounterparty] = Field(alias="topIssuedCounterparties")
    top_received_counterparties: List[TopCounterparty] = Field(alias="topReceivedCounterparties")

    class Config:
        populate_by_name = True
